File: bindings/dotnet/src/Wcl/Core/Diagnostic.cs
namespace Wcl.Core
{
    public class Diagnostic
    {
        public string Severity { get; }
        public string Message { get; }
        public string? Code { get; }

        public Diagnostic(string severity, string message, string? code = null)
        {
            Severity = severity;
            Message = message;
            Code = code;
        }

        public bool IsError => Severity == "error";

        public override string ToString() =>
            Code != null ? $"[{Code}] {Severity}: {Message}" : $"{Severity}: {Message}";
    }
}

File: bindings/jvm/src/main/java/io/github/wiltaylor/wcl/core/Diagnostic.java
package io.github.wiltaylor.wcl.core;

public record Diagnostic(String severity, String message, String code) {
    public boolean isError() {
        return "error".equals(severity);
    }

    @Override
    public String toString() {
        if (code != null) {
            return "[" + code + "] " + severity + ": " + message;
        }
        return severity + ": " + message;
    }
}
